userspace/libscap/scap.c

```diff
@@ -1236,50 +1236,56 @@ uint64_t scap_get_driver_schema_version(scap_t* handle)
 int32_t scap_get_boot_time(char* last_err, uint64_t *boot_time)
 {
 #ifdef __linux__
-	struct timespec ts_uptime = {0};
-	struct timespec tv_now = {0};
-	uint64_t now = 0;
-	uint64_t uptime = 0;
-	char proc_dir[PPM_MAX_PATH_SIZE];
-	struct stat targetstat = {0};
-
-	/* More reliable way to get boot time */
-	snprintf(proc_dir, sizeof(proc_dir), "%s/proc/1/", scap_get_host_root());
-	if (stat(proc_dir, &targetstat) == 0)
-	{
-		/* This approach is constant between agent re-boots */
-		*boot_time = targetstat.st_ctim.tv_sec * (uint64_t) SECOND_TO_NS + targetstat.st_ctim.tv_nsec;
-		return SCAP_SUCCESS;
-	}
-
-	/*
-	 * Fall-back method
+	uint64_t btime = 0;
```

Comment:
This LGTM!
In the future, we might want to move the `scap_get_boot_time` to the scap_platform lib (@gnosek @Andreagit97 ) to be able to implement it on osx and win too.

Reply:
I live in this future already ;) it's just that I have to keep fixing stuff after every rebase :<

Reply:
So, how living in the 3000 is?

Reply:
Not bad though :rofl: